File: lib/src/main/java/net/imoya/android/preference/view/PreferenceItemView.java
package net.imoya.android.preference.view;

import android.content.SharedPreferences;

/**
 * 設定項目ビュー共通メソッド定義
 */
public interface PreferenceItemView {
    /**
     * {@link SharedPreferences} 更新時の処理を行います。
     *
     * @param sharedPreferences 更新された {@link SharedPreferences}
     * @param key 更新された項目のキー
     * @return 処理を行った場合は true, そうでない場合は false
     */
    boolean onPreferenceChange(SharedPreferences sharedPreferences, String key);
}

File: lib/src/main/java/net/imoya/android/preference/view/PreferenceItemView.kt
/*
 * Copyright (C) 2022 IceImo-P
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.imoya.android.preference.view

import android.content.SharedPreferences

/**
 * 設定項目ビュー共通メソッド定義
 */
fun interface PreferenceItemView {
    /**
     * [SharedPreferences] 更新時の処理を行います。
     *
     * @param sharedPreferences 更新された [SharedPreferences]
     * @param key 更新された項目のキー
     * @return 処理を行った場合は true, そうでない場合は false
     */
    fun onPreferenceChange(sharedPreferences: SharedPreferences, key: String): Boolean
}